src/components/Modal/Modal.js

```diff
@@ -30,7 +41,8 @@ Modal.propTypes = {
   isOpen: PropTypes.bool,
   toggle: PropTypes.func,
   handleClick: PropTypes.func,
-  hasTimeout: PropTypes.bool
+  hasTimeout: PropTypes.bool,
+  SearchIcon: PropTypes.bool
```

Comment:
I think this name not correct? What about using `haveIcon` or similar? Also you can pass this "search" logic to children, as is too specific, but this component is general